src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -306,6 +307,40 @@ public void parseTutorials_duplicateInput_singleInstanceStored() throws Exceptio
         assertEquals(expectedTutorials, actualTutorials);
     }
 
+    @Test
+    public void parseTaskIndex_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseTaskIndex((String) null));
+    }
+
+    @Test
+    public void parseTaskIndex_invalidInputZero_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseTaskIndex(INVALID_TASK_INDEX_ZERO));
+    }
+
+    @Test
+    public void parseTaskIndex_invalidInputNegative_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseTaskIndex(INVALID_TASK_INDEX_NEGATIVE));
+    }
+
+    @Test
+    public void parseTaskIndex_invalidInputNonNumeric_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseTaskIndex(INVALID_TASK_INDEX_NON_NUMERIC));
+    }
+
+    @Test
+    public void parseTaskIndex_validInput_success() throws Exception {
+        int expectedTaskIndex = 1;
+        int actualTaskIndex = ParserUtil.parseTaskIndex("1");
+        assertEquals(expectedTaskIndex, actualTaskIndex);
+    }
+
+    @Test
+    public void parseTaskIndex_validInputWithWhitespace_success() throws Exception {
+        int expectedTaskIndex = 5;
+        int actualTaskIndex = ParserUtil.parseTaskIndex("  5  ");
```

Comment:
Follow how the previous withWhitespace test cases were already implemented

Reply:
alright!